Make revert and resume callable as action buttons

matsl opened the pull request:
# What

Let activities-resume take a string argument so it can be called
directly using an activity's name.

Similarly let activities-revert activity parameter be optional and
default to current activity.

# Why

Proposing a change within the current API instead of adding new
functions. See PR #149 .

matsl committed:
Make revert and resume callable as action buttons

Let activities-resume take a string argument so it can be called
directly using an activity's name.

Similarly let activities-revert activity parameter be optional and
default to current activity.

rswgnu commented on activities.el:
```diff
@@ -411,6 +411,10 @@ available."
+  (when (stringp activity)
+    (unless (activities-named activity)
+      (user-error "No activity: %s" activity))
+    (setq activity (activities-named activity)))
```
```emacs-lisp
  ;; Simplified change version:
  (when (stringp activity)
    (unless (setq activity (activities-named activity))
      (user-error "No activity: %s" activity)))
```

matsl committed:
Use activity-or-name for parameter

Be explicit in the parameter naming. Makes the type conversion from
string to activity clearer.

alphapapa commented:
Hi Mats,

Having thought about this for a while, I'd prefer not to modify Activities' APIs like this.  The API is consistent in taking an activity struct as the first argument to functions, and changing that would require making changes all over the codebase.  And I'd prefer not to introduce inconsistencies by only changing a few functions in that way, because that would also be confusing to users of the APIs, and it would add a lot of boilerplate-type code for little benefit.

You may be able to work around some of this by using advice in Hyperbole, for example:

```elisp
(with-eval-after-load 'activities
  (define-advice activities-revert (:filter-args (args) hyperbole--activities-revert-advice)
    (unless (car args)
      (setf (car args) (or (activities-current)
                           (user-error "No current activity"))))
    args))
```

However, you should be careful, as Activities itself applies some advice to some of its functions (see `activities-tabs-mode`), and the order in which the advice is applied and unapplied does matter.

rswgnu commented:
Mats and I will take care of this in Hyperbole then, integrating with activities.el as needed.  Thanks.

alphapapa commented:
Thanks, Bob.  I appreciate your wanting to integrate with Activities.  Please let me know if I can help in other ways.
